### PR TITLE
Move InMemoryStorageClient from panel-exchange-client

### PR DESCRIPTION
**Pull request by corbantek**
- Will follow up with a similar change removing from panel-exchange-client

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/common-jvm/38)

### CODE REVIEW

**Review by efoxepstein — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/common-jvm/38)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas, @stevenwarejones, and @wangyaopw)

**Review by wangyaopw — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/common-jvm/38)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas and @stevenwarejones)

File: src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  override val defaultBufferSizeBytes: Int
+    get() = BYTE_BUFFER_SIZE
+
+  override suspend fun createBlob(blobKey: String, content: Flow<ByteString>): StorageClient.Blob {
+    require(!storageMap.containsKey(blobKey)) { "Cannot write to an existing key: $blobKey" }
+
+    // As we're using this primarily for unit tests, we want to collect the input to record
+    // size and to emulate "writing out" to memory.
+    val newBlob = Blob(content.flatten(), blobKey)
+    storageMap[blobKey] = newBlob
```
Comment (SanjayVas) [inline src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt:50]: Note that you're using a `ConcurrentMap`, but not actually using its atomic `putIfAbsent` operation.

File: src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * The default byte buffer size. Chosen as it is a commonly used default buffer size in an attempt
+ * to keep the tests as close to actual usage as possible.
+ */
+private const val BYTE_BUFFER_SIZE = BYTES_PER_MIB * 1
+
+/** [StorageClient] for InMemoryStorage service. */
+class InMemoryStorageClient(
+  private var storageMap: ConcurrentHashMap<String, StorageClient.Blob> =
```
Comment (SanjayVas) [inline src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt:33]: This should be a `val` as its never reassigned.

File: src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  override fun getBlob(blobKey: String): StorageClient.Blob? {
+    return storageMap[blobKey]
+  }
+
+  private inner class Blob(private val byteData: ByteString, private val blobKey: String) :
+    StorageClient.Blob {
+
+    override val size: Long = byteData.size().toLong()
```
Comment (SanjayVas) [inline src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt:62]: nit: We have the underlying `byteData`, so no need to copy its value to a backing field.

```
override val size: Long
  get() = byteData.size().toLong()
```

File: src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    override val storageClient: InMemoryStorageClient = this@InMemoryStorageClient
```
Comment (SanjayVas) [inline src/main/kotlin/org/wfanet/measurement/storage/testing/InMemoryStorageClient.kt:64]: Same here. Define a getter()

**Review by SanjayVas — COMMENTED** [review]
Sorry, this was already merged so can't leave comments in Reviewable. Please create a follow-up PR to clean up.

**Pull request merged**